Sources/App/Models/Version.swift

```diff
@@ -41,6 +41,7 @@ final class Version: Model, Content {
     @Parent(key: "package_id")
     var package: Package
 
+    // TODO: sas-2020-04-30: Explore folding branch/rag into an enum with associated value String
     @Field(key: "branch_name")
     var branchName: String?
```

Comment:
I think this is quite an important thing to sort out to avoid having to do queries where need to check for either and/or add a constraint to ensure only one is set. Instead we could try and save to a `revision` field that maps to a `Revision` enum:

```
enum Revision {
  case branch(String)
  case tag(String)
  case commit(String) // unlikely we'll actually need/want this
}
```

Reply:
Sounds good, but I agree we won't need anything tied to a specific commit. Let's just do branch/tag. 👍

Reply:
I'm curious how an enum like this would be serialised to a database field. Can't wait to see :)